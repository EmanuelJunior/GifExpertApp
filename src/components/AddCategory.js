import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {

    const [inputValue, setInputValue] = useState(''); // String

    const handleInputChange = ( e ) => {
        setInputValue( e.target.value );
    }

    const handleSubmit = ( event ) => {
        event.preventDefault();

        if( inputValue.trim().length >= 2 ){
            setCategories( categories => [inputValue, ...categories] );
            setInputValue('');
            console.log('Llegamos hasta aca!');
        }
    }

    return (
        <form onSubmit={ handleSubmit }>
            <p>{ inputValue }</p>
            <input 
                type="text"
                value={ inputValue }
                // onChange es parecido a el evento del teclado keydown
                onChange={ handleInputChange }
            />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories : PropTypes.func.isRequired
}

