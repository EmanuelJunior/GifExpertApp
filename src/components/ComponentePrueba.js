import { useEffect } from 'react';
import { Fragment } from 'react';

const ComponentePrueba = () => {

    const saludar = () => alert("Hello world");
    saludar();

    return (
        <Fragment>
            <h1 className="titulo" id='titulo' contentEditable={true}>Hola mundo desde Colom</h1>
            <p id="parr1">Esto es el primer parrafo</p>
            <p id="parr2">Este es el segundo parrafo</p>
            <b>Hola estoy en bold</b>

            <div className='anidado'>
                <p>Un parrafo</p>
                <h3>Un subtitulo</h3>
                <hr></hr>
                
                <input 
                    type='text' 
                    placeholder='Users'
                    className='userInput' 
                    value='Aca debe de ir el usuario'
                />
            </div>
        </Fragment>
    );
}

export default ComponentePrueba;