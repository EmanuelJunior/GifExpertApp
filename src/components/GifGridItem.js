import React from 'react'

export const GifGridItem = ({ id, title, url }) => {
    
    return (
        <div className='card animate__animated animate__bounceInRight'>
            <img src={ url } alt={ title }/>
            <p> { title } </p>
        </div>
    )
}
