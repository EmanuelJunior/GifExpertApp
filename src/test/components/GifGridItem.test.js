import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas en <GifGridItem />', ()=> {
    
    const title = 'Un titulo';
    const url = 'https://localhost/algo.jpg';
    const wrapper = shallow( <GifGridItem title={ title } url={ url }/> );

    test('Debe de mostrar el componente correctamente', ()=> {
        expect( wrapper ).toMatchSnapshot();
    });

    test('Debe de tener un parrafo con el title', ()=>{
        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe( title );
    });

    test('Debe de tener la imagen igual al url y alt de los props', () => {
        
        const img = wrapper.find('img');

        expect( img.prop('src') ).toBe( url );
        expect( img.prop('alt') ).toBe( title );

    });

    test('Debe de tener animate__bounceInRight', () => {
        
        const div = wrapper.find('div');
        const className = div.prop('className');
        
        expect( className.includes('animate__bounceInRight') ).toBe( true );
    });
    
    

});
