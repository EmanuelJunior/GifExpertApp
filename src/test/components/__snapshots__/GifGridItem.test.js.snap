// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> Debe de mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__bounceInRight"
>
  <img
    alt="Un titulo"
    src="https://localhost/algo.jpg"
  />
  <p>
     
    Un titulo
     
  </p>
</div>
`;
